### PR TITLE
Major refactor of code to eliminate the need to provide apikey on every function call

### PR DESCRIPTION
**Pull request by dwrod**
This substantive PR changes all functions within the module to eliminate the need to call apikey as part of every function call. Instead users will need to add an environment variable named FMP_API_KEY which will automatically be read.

### CODE REVIEW

**Comment by psohn** [comment]
Not a contributor here but I'm wondering if this could be done with backwards compatibility in mind. E.g. keep `apikey` as an optional param and if it's not supplied, it takes the env variable.

**Comment by dwrod** [comment]
that's fair @psohn . I've moved on to a different project for now but will try and get this done so that everyone can make usre of the updates elsewhere. there were a number of broken/missing functions given changes on the financialmodelingprep end.

File: fmpsdk/alternative_data.py [hunk 1]
```diff
@@ -1,49 +1,48 @@
...
 
     List of symbols for COT.
-
-    :param apikey: Your API key.
+    :param apikey: Your API key. If not provided, the function will use the API_KEY from environment variables.
     :return: A list of dictionaries.
     """
     path = f"commitment_of_traders_report/list"
-    query_vars = {"apikey": apikey}
+    query_vars = {"apikey": apikey} if apikey else {"apikey": API_KEY}
```
Comment (psohn) [inline fmpsdk/alternative_data.py:18]:
```suggestion
    query_vars = {"apikey": apikey or API_KEY}
```
nit
should work and be a little more concise but doesn't really matter

File: fmpsdk/alternative_data.py [hunk 2]
```diff
@@ -1,49 +1,48 @@
...
 
 from .url_methods import __return_json_v4
 
+# Read API key from environment variable
+API_KEY = os.environ.get('FMP_API_KEY')
 
-def commitment_of_traders_report_list(
-    apikey: str,
-) -> typing.Optional[typing.List[typing.Dict]]:
+def commitment_of_traders_report_list(apikey: str = None) -> typing.Optional[typing.List[typing.Dict]]:
```
Comment (psohn) [inline fmpsdk/alternative_data.py:9]:
```suggestion
def commitment_of_traders_report_list(apikey: typing.Optional[str] = None) -> typing.Optional[typing.List[typing.Dict]]:
```

File: fmpsdk/alternative_data.py (hunk 1 above)
Comment (psohn) [reply fmpsdk/alternative_data.py:18]: also wondering (maybe not in the scope of this work) if an error should be raised if neither an `apikey` nor `API_KEY` are supplied

**Comment by vquelque** [comment]
+1 for merging this 🙏

**Pull request closed without merging**